components/Tabs/tests/unit/TabBarView/MDCTabBarViewTests.m

```diff
@@ -880,4 +880,58 @@ - (void)testAccessibilityElementForItemInItemsArrayReturnsItemViewWithMatchingTi
   }
 }
 
+- (void)testRectForItemNotFoundReturnsNullRectangle {
+  // Given
+  self.tabBarView.items = @[];
+
+  // When
+  CGRect itemFrame = [self.tabBarView rectForItem:self.itemC inCoordinateSpace:self.tabBarView];
+
+  // Then
+  XCTAssertTrue(CGRectIsNull(itemFrame), @"(%@) is not equal to (%@)",
+                NSStringFromCGRect(itemFrame), NSStringFromCGRect(CGRectNull));
+}
+
+- (void)testRectForItemConvertedToTabBarView {
+  // Given
+  self.tabBarView.items = @[ self.itemA ];
+  [self.tabBarView sizeToFit];
+  [self.tabBarView layoutIfNeeded];
+
+  // When
+  CGRect itemFrame = [self.tabBarView rectForItem:self.itemA inCoordinateSpace:self.tabBarView];
+
+  // Then
+  XCTAssertTrue(CGRectEqualToRect(itemFrame, self.tabBarView.bounds), @"(%@) is not equal to (%@)",
+                NSStringFromCGRect(itemFrame), NSStringFromCGRect(self.tabBarView.bounds));
+}
+
+- (void)testRectForItemConvertedToSuperView {
+  // Given
+  UIOffset tabBarOffsetWithinSuperview = UIOffsetMake(30, 40);
+  self.tabBarView.items = @[ self.itemA ];
+  [self.tabBarView sizeToFit];
+  [self.tabBarView layoutIfNeeded];
+  UIView *superView =
+      [[UIView alloc] initWithFrame:CGRectMake(0, 0,
+                                               CGRectGetWidth(self.tabBarView.bounds) +
+                                                   tabBarOffsetWithinSuperview.horizontal,
+                                               CGRectGetHeight(self.tabBarView.bounds) +
+                                                   tabBarOffsetWithinSuperview.vertical)];
+  [superView addSubview:self.tabBarView];
+  self.tabBarView.center =
```

Comment:
Rather than this, could we just use CGRectOffset? That might be more clear rather than trying to mess with the center of the element in order to offset it.

Reply:
I'm probably missing something. My understanding is that in general it's better to manipulate `bounds` and `center` rather than `frame` on a view. That's because the `frame` is undefined when the transform is anything other than the identity.  I was using `center` here because it was rather simple and didn't require me to standardize the `frame` as an initial step.

Reply:
`bounds` and `center` manipulate the frame. they are syntactic sugar/convenience.

I imagine you should probably make a local variable to copy the CGRect of the frame, manipulate it, and then assign it back. (Multiple accesses of the `frame` are also suboptimal)

Reply:
I agree that multiple accesses are suboptimal and could be improved. I'll pull the frame into a local variable.

It is my understanding that `bounds` and `center` are not syntactic sugar for `frame`, nor convenience. I believe that the opposite is actually true.  Based on [Apple's documentation for `bounds`](https://developer.apple.com/documentation/uikit/uiview/1622580-bounds) it seems that it and `center` are intended as the source of the value of `frame` (potentially along with `transform`).  The [documentation for the `center` property](https://developer.apple.com/documentation/uikit/uiview/1622627-center) explicitly calls-out that property as the correct way to move a view within its superview's bounds.

>  Use this property, **instead of the frame property**, when you want to change the position of a view. The center point is always valid, even when scaling or rotation factors are applied to the view's transform (emphasis added). 

I think [the UIKit documentation for `frame`](https://developer.apple.com/documentation/uikit/uiview/1622621-frame) makes it pretty clear that using `center` and `bounds` can be preferred over `frame` when the caller is unsure if the `transform` has been manipulated.

> . . . if the transform property contains a non-identity transform, the value of the frame property is undefined and should not be modified. In that case, **reposition the view using the center property and adjust the size using the bounds property instead** (emphasis added).

In general, I do personally prefer manipulating `center` and `bounds` rather than `frame`, but it often involves arithmetic "gymnastics" that make the code harder to read.  Since `transform` manipulations are rare, it's often safe to go with `frame` as a shortcut.

Reply:
Interesting. I hadn't read all the way through the documentation. I stopped after the line that read: 

> Setting this property updates the origin of the rectangle in the frame property appropriately.

If Apple recommends using those fields instead, I have no objection to it.

Reply:
Leaving it for now.